src/offset_date_time.rs

```diff
@@ -1320,4 +1332,29 @@ impl From<OffsetDateTime> for SystemTime {
         }
     }
 }
+
+#[allow(clippy::fallible_impl_from)]
+#[cfg(all(
+    target_arch = "wasm32",
+    not(any(target_os = "emscripten", target_os = "wasi"))
+))]
+impl From<js_sys::Date> for OffsetDateTime {
+    fn from(js_date: js_sys::Date) -> Self {
+        let timestamp = (js_date.get_time() / 1000.0) as i64;
+        Self::from_unix_timestamp(timestamp)
+            .expect("invalid timestamp: Timestamp cannot fit in range")
+    }
+}
+
+#[cfg(all(
+    target_arch = "wasm32",
+    not(any(target_os = "emscripten", target_os = "wasi"))
+))]
+impl From<OffsetDateTime> for js_sys::Date {
+    fn from(datetime: OffsetDateTime) -> Self {
+        let timestamp = datetime.unix_timestamp() as f64;
+        js_sys::Date::new(&timestamp.into())
```

Comment:
In a similar manner, this needs to use `.unix_timestamp_nanos()`. Without that you are discarding subsecond information. The nanoseconds can be divided by 1,000,000 before being cast to `f64`.